--- libcaca-0.99.beta19/configure.ac.old	2021-03-02 18:37:24.803307367 -0600
+++ libcaca-0.99.beta19/configure.ac	2021-03-02 18:41:42.708061480 -0600
@@ -235,12 +235,14 @@
   AC_CHECK_HEADERS(GL/gl.h OpenGL/gl.h,
    [AC_CHECK_HEADERS(GL/glut.h,
      [AC_CHECK_LIB(glut, glutCloseFunc,
-       [AC_DEFINE(HAVE_GLUTCLOSEFUNC, 1, Define to 1 if you have the ‘glutCloseFunc’ function.)])
+       [AC_DEFINE(HAVE_GLUTCLOSEFUNC, 1, Define to 1 if you have the ‘glutCloseFunc’ function.)],
+       [],-lGL)
       AC_CHECK_LIB(glut, glutMainLoopEvent,
-       [ac_cv_my_have_gl="yes"])
+       [ac_cv_my_have_gl="yes"],[ac_cv_my_have_gl="no"],-lGL)
       AC_CHECK_LIB(glut, glutCheckLoop,
        [ac_cv_my_have_gl="yes"
-        AC_DEFINE(HAVE_GLUTCHECKLOOP, 1, Define to 1 if you have the ‘glutCheckLoop’ function.)])])
+        AC_DEFINE(HAVE_GLUTCHECKLOOP, 1, Define to 1 if you have the ‘glutCheckLoop’ function.)],
+       [],-lGL)])
       break])
   if test "${ac_cv_my_have_gl}" = "yes"; then
     AC_DEFINE(USE_GL, 1, Define to 1 to activate the OpenGL backend driver)
