--- dracut-048/install/dracut-install.c.orig	2018-07-06 10:37:51.000000000 +0200
+++ dracut-048/install/dracut-install.c	2019-10-10 13:39:46.544968901 +0200
@@ -1243,6 +1243,9 @@
                 mod = kmod_module_get_module(itr);
                 path = kmod_module_get_path(mod);
 
+		if (path == NULL)
+			continue;
+
                 name = kmod_module_get_name(mod);
                 if (arg_mod_filter_noname && (regexec(&mod_filter_noname, name, 0, NULL, 0) == 0)) {
                         kmod_module_unref(mod);
