--- framebuffer/Makefile.target.orig	2015-03-31 08:18:28.589624583 +0200
+++ framebuffer/Makefile.target	2015-03-31 08:21:00.422663635 +0200
@@ -48,6 +48,7 @@
 $(eval $(call feature_enabled,WEBP,-DWITH_WEBP,-lwebp,WebP (libwebp)))
 $(eval $(call pkg_config_find_and_add_enabled,MOZJS,mozjs185,JavaScript))
 $(eval $(call pkg_config_find_and_add_enabled,JS,mozilla-js,JavaScript))
+$(eval $(call pkg_config_find_and_add_enabled,VIDEO,gstreamer-0.10,Video))
 
 CFLAGS += -std=c99 -g -Dsmall $(WARNFLAGS) 	\
 	  -D_BSD_SOURCE \
