--- ./drivers/bluetooth/hci_bcm.c.orig	2017-06-09 00:00:00.000000000 +0200
+++ ./drivers/bluetooth/hci_bcm.c	2017-06-09 00:00:00.000000000 +0200
@@ -865,6 +865,7 @@
 	{ "BCM2E3D", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
 	{ "BCM2E3F", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
 	{ "BCM2E40", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
+	{ "BCM2E47", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
 	{ "BCM2E54", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
 	{ "BCM2E55", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
 	{ "BCM2E64", (kernel_ulong_t)&acpi_bcm_int_last_gpios },
