--- ./modules.d/90crypt/parse-crypt.sh.orig	2021-05-27 14:34:19.000000000 +0200
+++ ./modules.d/90crypt/parse-crypt.sh	2021-12-29 08:21:26.029440752 +0100
@@ -166,7 +166,7 @@
             if [ $is_keysource -eq 0 ]; then
                 uuid=$luksid
                 while [ "$uuid" != "${uuid#*-}" ]; do uuid=${uuid%%-*}${uuid#*-}; done
-                printf -- '[ -e /dev/disk/by-id/dm-uuid-CRYPT-LUKS?-*%s*-* ] || exit 1\n' "$uuid" \
+		printf -- '[ -e /dev/disk/by-id/dm-uuid-CRYPT-LUKS?-*%s*-* -o -e /dev/mapper/%s ] || exit 1\n' "$uuid" "$luksname" \
                     >> "$hookdir/initqueue/finished/90-crypt.sh"
                 {
                     printf -- '[ -e /dev/disk/by-uuid/*%s* ] || ' "$luksid"
