--- proton-bridge-br-1.6.3/internal/frontend/qt/frontend_nogui.go.orig	2021-02-10 14:45:38.000000000 +0100
+++ proton-bridge-br-1.6.3/internal/frontend/qt/frontend_nogui.go	2021-02-21 10:07:28.225042419 +0100
@@ -37,11 +37,11 @@
 type FrontendHeadless struct{}
 
 func (s *FrontendHeadless) Loop() error {
-	log.Info("Check status on localhost:8081")
+	log.Info("Check status on localhost:8180")
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "Bridge is running")
 	})
-	return http.ListenAndServe(":8081", nil)
+	return http.ListenAndServe(":8180", nil)
 }
 
 func (s *FrontendHeadless) NotifyManualUpdate(update updater.VersionInfo, canInstall bool) {
