--- ./ui/base/dragdrop/os_exchange_data_provider_aurax11.h.orig	2014-02-01 12:16:13.749288587 -0800
+++ ./ui/base/dragdrop/os_exchange_data_provider_aurax11.h	2014-02-01 12:15:37.112624173 -0800
@@ -65,7 +65,8 @@
   virtual void SetPickledData(const OSExchangeData::CustomFormat& format,
                               const Pickle& pickle) OVERRIDE;
   virtual bool GetString(base::string16* data) const OVERRIDE;
-  virtual bool GetURLAndTitle(GURL* url, base::string16* title) const OVERRIDE;
+  virtual bool GetURLAndTitle(OSExchangeData::FilenameToURLPolicy policy,
+      GURL* url, base::string16* title) const OVERRIDE;
   virtual bool GetFilename(base::FilePath* path) const OVERRIDE;
   virtual bool GetFilenames(
       std::vector<OSExchangeData::FileInfo>* filenames) const OVERRIDE;
