--- upower-0.99.3/src/linux/up-device-supply.c.orig	2017-03-04 10:26:18.818585233 +0100
+++ upower-0.99.3/src/linux/up-device-supply.c	2017-03-04 10:44:39.528679051 +0100
@@ -1010,6 +1010,7 @@
 	const gchar *native_path;
 	const gchar *scope;
 	UpDeviceKind type;
+	gchar *model_name = NULL;
 
 	up_device_supply_reset_values (supply);
 
@@ -1039,6 +1040,15 @@
 		return FALSE;
 	}
 
+	/* ignore Logitech diNovo Edge (Wireless Keyboard) -- as the battery reading is wrong */
+	model_name = up_device_supply_get_string (native_path, "model_name");
+	if (model_name != NULL) {
+		if (g_ascii_strcasecmp (model_name, "Logitech diNovo Edge") == 0) {
+			g_debug ("Ignoring Logitech diNovo Edge, as the battery reading is wrong");
+			return FALSE;
+		}
+	}
+
 	/* try to detect using the device type */
 	type = up_device_supply_guess_type (native, native_path);
 
